### PR TITLE
Fix CA generation, remove env API usage

### PR DESCRIPTION
**Pull request by kirgene**

### CODE REVIEW

File: src/lib/vault/index.js [hunk 1]
```diff
@@ -94,75 +98,120 @@ class Vault {
...
+    /**
+     * Create root CA
+     * @param {Object} caOptions
+     */
+    async createCA (caOptions) {
+        // eslint-disable-next-line no-empty
+        try { await this.deleteCA(); } catch (e) { }
+
+        const { names } = caOptions.csr;
+        const altNames = names.length > 1 ? names.slice(1).map((name) => name.CN).join(',') : '';
```
Comment (partiallyordered) [inline src/lib/vault/index.js:174]: _Might_ be clearer as:
```suggestion
        const [,...altNamesObjs] = names;
        const altNames = altNamesObjs.map((name) => name.CN).join(',');
```

File: src/lib/vault/index.js [hunk 2]
```diff
@@ -94,75 +98,120 @@ class Vault {
...
+            key: data.private_key,
+        };
     }
 
-    async signCSR(params) {
+    async createDFSPServerCert (csrParameters) {
+        const reqJson = {
+            common_name: csrParameters.subject.CN,
+        };
+        if (csrParameters.extensions && csrParameters.extensions.subjectAltName) {
```
Comment (partiallyordered) [inline src/lib/vault/index.js:202]:
```suggestion
        if (csrParameters?.extensions?.subjectAltName) {
```

File: src/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "mojaloop-payment-manager-management-api-service",
-  "version": "2.0.4",
+  "version": "2.0.5",
```
Comment (partiallyordered) [inline src/package.json:3]: Hasn't the external API had breaking changes? If so, shouldn't this be `3.0.0`?

**Review by partiallyordered — COMMENTED** [review]
It looks like perhaps the version of `@modusbox/mcm-client` needs to be updated. It appears many calls to this lib in the `1.0.34` version required in the `package.json` still expect the `envId` arg. I suppose this PR #8 depends on https://github.com/pm4ml/mcm-client/pull/1 being merged. This PR should probably be converted to a draft until said PR is merged. Or the version of `@modusbox/mcm-client` modified to the anticipated version (even if that does not yet exist) to make this clear.

I also noticed during the review that some of the `operationId` values in the API spec no longer exist. If this is not how requests are mapped to handlers, this should not be important. However, it might therefore pay to remove all `operationId` values in the API spec, unless they serve some other purpose.

File: src/UIAPIServer/api.yaml [hunk 4]
```diff
@@ -588,27 +524,6 @@ paths:
...
-            schema:
-              $ref: '#/components/schemas/BaseJWSCert'
-      responses:
-        "200":
-          description: successful operation
-          content:
-            application/json:
-              schema:
-                $ref: "#/components/schemas/JWSCert"
-      x-swagger-router-controller: DfspJWSCerts
```
[the remark this answers is not in the document]
Comment (kirgene) [reply src/UIAPIServer/api.yaml:611]: Yes, JWS keys are no longer manually updated

File: src/lib/model/CertificatesModel.js [hunk 5]
```diff
@@ -36,152 +35,189 @@ class CertificatesModel {
...
+
+    async storeJWS(keypair) {
+        await this._vault.setJWS(keypair);
+        await this._connectorManager.reconfigureOutboundSdkForJWS(keypair.privateKey);
+    }
+
+    validateJWSKeyPair(jwsKeyPair) {
+        forge.pki.publicKeyFromPem(jwsKeyPair.publicKey);
+        forge.pki.privateKeyFromPem(jwsKeyPair.privateKey);
+    }
```
[the remark this answers is not in the document]
Comment (kirgene) [reply src/lib/model/CertificatesModel.js:194]: We should return the error message to a user, so can't use this approach.

File: src/lib/model/MCMStateModel.js [hunk 6]
```diff
@@ -98,19 +68,24 @@ class MCMStateModel {
...
-        const cache = this._db.redisCache;
-        const inboundEnrollmentId = await cache.get(`inboundEnrollmentId_${this._envId}`);
-        this._logger.log(`inboundEnrollmentId: ${inboundEnrollmentId}`);
-        if (inboundEnrollmentId) {
-            const privateKey = await this._vault.getClientPrivateKey();
+        try {
+            await this._certificatesModel.exchangeOutboundSdkConfiguration();
+        } catch (err) {
+            this._logger.log(`Error refreshing client certificate: ${err.stack || util.inspect(err)}`);
+        }
```
[the remark this answers is not in the document]
Comment (kirgene) [reply src/lib/model/MCMStateModel.js:75]: This function is used in the "refresh" method and shouldn't throw even if there are errors.

**Comment by kirgene** [comment]
Regarding `operationId`: I had a plan to refactor API handlers to use https://www.npmjs.com/package/oas3-tools

**Review by partiallyordered — APPROVED** [review]
Looks good. I have a question: are we sure there's no longer any dependency on cfssl? I recall removing that from a Dockerfile in the past and having problems as a result. It may not have even been this service. I have a hazy memory of a transitive cfssl dependency. One of the dependencies that generated certs shelled out to cfssl and when I removed it from the Dockerfile I broke stuff. Anyway, I'm pretty sure you've addressed this, I thought I'd mention it to be sure. But I've approved because I expect the result of this will be "yes Matt, that's been addressed; merge".

**Comment by kirgene** [comment]
cfssl is no longer used, as we now use Vault and node-forge for cert/secrets generation.

**Pull request merged**